internal/sqlutil/migrate.go

```diff
@@ -50,6 +50,7 @@ type Migration struct {
 
 // Migrator
 type Migrator struct {
+	gooseMutex      sync.Mutex
```

Comment:
Do you see any races without this mutex?
There's already a mutex on this struct, which is protecting `knownMigrations`, the other maps shouldn't be accessed concurrently.

Reply:
Yes, the race is coming from accessing goose in two separate goroutines, I'll add a comment clarifying the need for this lock.

Reply:
Oh, I just saw that this PR was created before #2264 was merged. So I _think_ we may not need this mutex at all.

Reply:
Ah, nice, thank you, I missed this. I removed the mutex.